kep/9-capacity-scheduling/README.md

```diff
@@ -0,0 +1,324 @@
+scheduler extension to support the "ElasticQuota" based scheduling based on 
+the scheduler framework.
+
+[ResourceQuota]: https://kubernetes.io/docs/concepts/policy/resource-quotas/
+[Yarn capacity scheduler]: https://www.netjstech.com/2018/04/capacity-scheduler-in-yarn-hadoop.html
+
+### Goals
+
+- The API definition of ElasticQuota
+- Implement a scheduler plugin to honor the API of elastic resource quota.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+A "CRD" is needed to interact with the scheduler. The CRD kind name is 
+subject to change. We use "ElasticQuota" in the proposal. The CRD defines the
+min and max allocation for the managed resources such as cpu, memory or 
+extended resources. "ElasticQuota" is namespace scoped. 
+
+```go
+// ElasticQuota sets elastic quota restrictions per namespace
+type ElasticQuota struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+	Spec   ElasticQuotaSpec
+	Status ElasticQuotaStatus
+}
+
+// ElasticQuotaSpec defines the Min and Max for Quota.
+type ElasticQuotaSpec struct {
+	Min v1.ResourceList
+	Max v1.ResourceList
+}
+
+// ElasticQuotaStatus defines the observed use.
+type ElasticQuotaStatus struct {
+	Min  v1.ResourceList
+	Max  v1.ResourceList
+	Used v1.ResourceList
+}
+```
+
+sample yaml is listed below:
+
+```yaml
+apiVersion: v1
+kind: ElasticQuota
+metadata:
+  name: test
+  namespace: test
+spec:
+  max:
+    cpu: 20
+    memory: 40Gi
+    nvidia.com/gpu: 2
+  min:
+    cpu: 10
+    memory: 20Gi
+    nvidia.com/gpu: 1
+```
+
+The definitions of "min" and "max" have been explained above. Max is infinite
+by default, min is 0 by default. The sum of min in all ElasticQuota must be 
+less than or equal to the total resource capacity of the cluster.The 
+ElasticQuota objects are created by a cluster administrator. One ElasticQuota
+object per namespace.
+
+### Relationship with ResourceQuota
+This version of proposal does not discuss merging the ElasticQuota into the 
+existing ResourceQuota. But in the future, we will try to merge ElasticQuota 
+into ResourceQuota when it's mature. We have three fields of resource in 
+ElasticQuota(min max) and ResourceQuota(hard). They act on different phases 
+of pod lifecycle. So it maybe possible to merge them together.
+- hard: it will check the resource in pod creation through an admission check
+. It will prevent submitting too many jobs at one time. The usage of hard is 
+aggregated based on the resource configurations of all pods so it will be 
+greater than or equal to max. 
+
+- max: it will check the resource in the Prefilter of Pod Scheduling Cycle. 
+The usage of max is aggregated based on the resource configurations of 
+successfully scheduled pods.
+
+- min: it means the guaranteed resource of Quota.
+
+min and max need to respect the namespace's ResourceQuota limit, i.e., `min 
+<= max <= hard` . 
```

Comment:
The relationship need to be established [per PriorityClass if they are so specified](https://kubernetes.io/docs/concepts/policy/resource-quotas/#resource-quota-per-priorityclass), right?

This would imply that `ElasticQuota` also need to support priority classes?